api/api_test.go

```diff
@@ -0,0 +1,538 @@
+package api
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/arhsxro/platform-go-challenge/models"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+)
+
+type MockStore struct {
+	GetUserFavoritesFunc  func(ctx context.Context, userID, filterType string, page, pageSize int) ([]models.Asset, error)
+	AddFavoriteFunc       func(ctx context.Context, userID string, asset models.Asset) error
+	RemoveFavoriteFunc    func(ctx context.Context, userID, assetId string) error
+	UpdateDescriptionFunc func(ctx context.Context, userID, assetID, newDescription string) error
```

Comment:
Nice way of mocking!

Reply:
Thank you for the feedback!The truth is I spend a lot of time in tests.